ros2param/ros2param/verb/set.py

```diff
@@ -37,14 +37,29 @@ def add_arguments(self, parser, cli_name):  # noqa: D102
             'node_name', help='Name of the ROS node')
         arg.completer = NodeNameCompleter(
             include_hidden_nodes_key='include_hidden_nodes')
-        parser.add_argument(
-            '--include-hidden-nodes', action='store_true',
-            help='Consider hidden nodes as well')
+
         arg = parser.add_argument(
-            'parameter_name', help='Name of the parameter')
+            'parameters', nargs='*',
+            help='List of parameter name and value pairs i.e. "int_param 1 str_param hello_world"')
         arg.completer = ParameterNameCompleter()
+
         parser.add_argument(
-            'value', help='Value of the parameter')
+            '--include-hidden-nodes', action='store_true',
+            help='Consider hidden nodes as well')
+
+    def build_parameters(self, params):
+        parameters = []
+        if len(params) % 2:
+            raise RuntimeError('Must pass list of parameter name and value pairs')
+
+        params = [(params[i], params[i+1]) for i in range(0, len(params), 2)]
+        for param_str in params:
+            parameter = Parameter()
+            parameter.name = param_str[0]
+            parameter.value = get_parameter_value(string_value=param_str[1])
+            parameters.append(parameter)
```

Comment:
This seems like we are iterating over the list twice for no really good reason.  This can be simplified to:

```suggestion
        for i in range(0, len(params), 2):
            parameter = Parameter()
            parameter.name = params[i]
            parameter.value = get_parameter_value(string_value=params[i+1])
            parameters.append(parameter)
```